compiler/rustc_infer/src/infer/outlives/verify.rs

```diff
@@ -282,15 +279,15 @@ impl<'cx, 'tcx> VerifyBoundCx<'cx, 'tcx> {
     ///
     /// then this function would return `'x`. This is subject to the
     /// limitations around higher-ranked bounds described in
-    /// `region_bounds_declared_on_associated_item`.
-    fn declared_projection_bounds_from_trait(
+    /// `declared_region_bounds`.
+    #[instrument(level = "debug", skip(self))]
+    pub fn bounds(
```

Comment:
I find the previous name more descriptive. Maybe we can generalize it to `declared_bounds_from_(trait|definition)`.

Perhaps it's good to inline `declared_region_bounds` too?